### PR TITLE
Add `architectures` support to prowgen

### PR DESCRIPTION
**Pull request by howardjohn**
I originally wanted to use the `matrix` feature. However, given how
    common this will be, I think it makes sense to have a specialized field.
    This makes it easier to apply some of the bespoke customizations we
    want.

For https://github.com/istio/istio/issues/26652#issuecomment-1155682106

Commit 1 is the main change, I recommend reviewing it. The second commit just generates an architecture node selector for each. This should have no impact on any existing jobs but will be needed for when we actually use multi-arch

### CODE REVIEW

**Review by fejta — APPROVED** [review]

File: tools/prowgen/pkg/testdata/simple.yaml [hunk 1]
```diff
@@ -44,9 +44,20 @@ jobs:
     resources: custom
     command: [prow/command.sh]
     repos: [istio/istio]
-    modifier: [presubmit_skipped]
```
Comment (ericvn) [inline tools/prowgen/pkg/testdata/simple.yaml:47]: Not sure if there was a reason to remove this line, but noting it was removed in the test-data as well.

File: tools/prowgen/pkg/testdata/simple.yaml [hunk 2]
```diff
@@ -44,9 +44,20 @@ jobs:
...
+    architectures: [amd64, arm64]
+    command: [prow/command.sh]
+    image: test
+
+  - name: multi-arch-param
+    types: [presubmit]
+    architectures: [amd64, arm64]
+    command: [prow/command.sh, $(params.arch)]
+    image: test
+
```
Comment (ericvn) [inline tools/prowgen/pkg/testdata/simple.yaml:60]: Should we have an arm64 only test job?

File: tools/prowgen/pkg/testdata/simple.gen.yaml [hunk 3]
```diff
@@ -332,6 +339,7 @@ presubmits:
           name: build-cache
           subPath: gocache
       nodeSelector:
+        kubernetes.io/arch: arm64
```
Comment (ericvn) [inline tools/prowgen/pkg/testdata/simple.gen.yaml]: Should this one be an amd64?

File: tools/prowgen/pkg/testdata/simple.yaml (hunk 1 above)
Comment (howardjohn) [reply tools/prowgen/pkg/testdata/simple.yaml:47]: it no longer is a real field - I made the yaml parser strict

File: tools/prowgen/pkg/testdata/simple.yaml (hunk 2 above)
Comment (howardjohn) [reply tools/prowgen/pkg/testdata/simple.yaml:60]: This means "make an arm job and make an amd job". So it's mostly syntax sugar - they are independent

File: tools/prowgen/pkg/testdata/simple.gen.yaml (hunk 3 above)
Comment (howardjohn) [reply tools/prowgen/pkg/testdata/simple.gen.yaml]: there should be one arm one per the job config

Comment (ericvn) [reply tools/prowgen/pkg/testdata/simple.gen.yaml]: Correct, and I believe I am seeing one arm and one amd for the second job config. But I am seeing that the first job config yielded two arm64 jobs (line 376 and line 342 (342 should be amd)) which would be incorrect.

Comment (howardjohn) [reply tools/prowgen/pkg/testdata/simple.gen.yaml]: Good catch! was a bug, fixed in last commit

**Review by ericvn — APPROVED** [review]

**Pull request merged**